node/network/collator-protocol/src/collator_side.rs

```diff
@@ -467,7 +495,16 @@ async fn advertise_collation(
 		(Some(collation), true) => collation.status.advance_to_advertised(),
 	}
 
-	let wire_message = protocol_v1::CollatorProtocolMessage::AdvertiseCollation(relay_parent, collating_on);
+	let advertise_collation_signature_payload =
+		protocol_v1::advertise_collation_signature_payload(&relay_parent, &collating_on);
+
+	let wire_message = protocol_v1::CollatorProtocolMessage::AdvertiseCollation(
+		relay_parent,
+		collating_on,
+		state
+			.collator_pair
+			.sign(&advertise_collation_signature_payload),
```

Comment:
Why do we sign `AdvertiseCollation` messages as well? We already reject messages if we have not received a `Declare` before and if we did, the collator has already proven its identity.

Reply:
Good point, will update the PR.